### DISCUSSION TITLE
Does velero take snapshot for the PV when begin backup a Pod?

### DISCUSSION THREAD

**Discussion by Deardrops** [vmware-tanzu/velero#3308] [Community support Q&A] [answered]
For getting a consistent view, does velero take snapshot for the PV when begin backup a Pod?
And then restic backups from this PV snapshot, instead of backuping the mounted PV in the running Pod.

For example, If I
1. Create a Pod with a PV
2. Write some data to the PV
3. Backup this Pod, meanwhile, write some data to the PV.

After step 3, Is the backup data create by restic represents the point in time when the backup started，
or the window of time from when the backup began to when it ended?

**Comment by Deardrops** [comment, marked as answer]
@skriss Could you take a look at this question? thanks

**Reply by dsu-igeek** [reply]
There are two different paths for PV backup at the moment.

Restic path - Restic is run against the volume, there is no ability for Restic to lock out applications from writing.  If the application writes to the volume while Restic is copying data, those changes may or may not be picked up depending on whether or not Restic has already backed up the files written.  So the backup data created by Restic can be inconsistent.  Using an execution hook to perform an fsfreeze (https://velero.io/docs/v1.5/backup-hooks/#hook-example-with-fsfreeze) will force the Restic backup to be consistent with when the fsfreeze was performed.

Volume snapshotter path - The volume snapshotters use the native snapshotting capabilities of the storage platform (e.g. AWS EBS) to take a backup.  That snapshot will be a crash-consistent point-in-time of when the snapshot was performed.
